packages/flutter/lib/src/material/dropdown_menu.dart

```diff
@@ -314,6 +316,8 @@ class _DropdownMenuState<T> extends State<DropdownMenu<T>> {
   void didUpdateWidget(DropdownMenu<T> oldWidget) {
     super.didUpdateWidget(oldWidget);
     if (oldWidget.dropdownMenuEntries != widget.dropdownMenuEntries) {
+      filteredEntries = widget.dropdownMenuEntries;
```

Comment:
Should currentHighlight be set to null here?

Reply:
Ah yes, and I think I need to add a if statement to check `oldWidget.enableSearch != widget.enableSearch` so that if `widget.enableSearch` is updated to false, `currentHighlight` should also be updated. Just updated.

```
if (oldWidget.enableSearch != widget.enableSearch) {
  if (!widget.enableSearch) {
    currentHighlight = null;
  }
}
```